griptape/events/event_bus.py

```diff
@@ -28,24 +32,21 @@ def remove_event_listeners(self, event_listeners: list[EventListener]) -> None:
             self.remove_event_listener(event_listener)
 
     def add_event_listener(self, event_listener: EventListener) -> EventListener:
-        with self._thread_lock:
-            if event_listener not in self._event_listeners:
-                self._event_listeners.append(event_listener)
+        if event_listener not in self.event_listeners:
+            self.event_listeners = self.event_listeners + [event_listener]
```

Comment:
Just curious, why don't we use a `set` instead of a `list` so we get uniqueness for free? Doesn't seem like we are relying on order for anything.

Reply:
I think just an initial oversight. We can update but I'd prefer to do so outside of this PR since there's enough complexity as is.